spitfire/runtime/udn.py

```diff
@@ -134,23 +135,31 @@ def _resolve_udn_prefer_attr3(_object, name, raise_exception=False):
 def _resolve_placeholder(name, template, global_vars):
     placeholder_cache = template.placeholder_cache
     if placeholder_cache and name in placeholder_cache:
-        return placeholder_cache[name]
+        ph = placeholder_cache[name]
+        if isinstance(ph, weakref.ReferenceType) and ph() is not None:
```

Comment:
Yeah, I think yours is prettier, too. Updating.

When ph() is None it means whatever was in the cache is no longer referenced by anything else and has been collected. In this case I allow execution to fall through to see if maybe there's a new object behind the same placeholder name.